// adal-angular8/adal8-angular.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var adal8_interceptor_1 = require("./adal8-interceptor");
var adal8_user_1 = require("./adal8-user");
var adal8_service_1 = require("./adal8.service");
var adal8_http_service_1 = require("./adal8-http.service");
var core_1 = require("@angular/core");
var http_1 = require("@angular/common/http");
var Adal8AgnularModule = /** @class */ (function () {
    function Adal8AgnularModule() {
    }
    Adal8AgnularModule = __decorate([
        core_1.NgModule({
            imports: [],
            exports: [
                adal8_user_1.Adal8User, adal8_service_1.Adal8Service, adal8_http_service_1.Adal8HTTPService, adal8_interceptor_1.Adal8Interceptor
            ],
            providers: [
                {
                    provide: http_1.HTTP_INTERCEPTORS,
                    useClass: adal8_interceptor_1.Adal8Interceptor,
                    multi: true
                }
            ]
        })
    ], Adal8AgnularModule);
    return Adal8AgnularModule;
}());
exports.Adal8AgnularModule = Adal8AgnularModule;
